src/pages/dev/reference/interchain-token-service-rate-limit.mdx

```diff
@@ -0,0 +1,193 @@
+    
+```solidity
+/**
+ * @notice Returns the current flow in amount.
+ * @return flowInAmount_ The current flow in amount.
+ */
+function flowInAmount() external view returns (uint256 flowInAmount_) {
+    uint256 epoch = block.timestamp / EPOCH_TIME;
+    uint256 slot = _getFlowInSlot(epoch);
+
+    assembly {
+        flowInAmount_ := sload(slot)
+    }
+}
+```
+    
+## Internal Management Functions
+
+The `FlowLimit` contract contains the following internal functions.
+
+### `_setFlowLimit(uint256, bytes32)`
+
+Sets the flow limit for a specific token.
+    
+```solidity
+/**
+ * @notice Internal function to set the flow limit.
+ * @param flowLimit_ The value to set the flow limit to.
+ * @param tokenId The id of the token to set the flow limit for.
+ */
+function _setFlowLimit(uint256 flowLimit_, bytes32 tokenId) internal {
+    assembly {
+        sstore(FLOW_LIMIT_SLOT, flowLimit_)
+    }
+
+    emit FlowLimitSet(tokenId, msg.sender, flowLimit_);
+}
+```
+    
+### `_addFlowOut(uint256)`
+
+Adds to the flow out amount, ensuring it does not exceed the flow limit.
+    
+```solidity
+/**
+ * @notice Adds a flow out amount.
+ * @param flowOutAmount_ The flow out amount to add.
+ */
+function _addFlowOut(uint256 flowOutAmount_) internal {
+    uint256 flowLimit_ = flowLimit();
+    if (flowLimit_ == 0) return;
+
+    uint256 epoch = block.timestamp / EPOCH_TIME;
+    uint256 slotToAdd = _getFlowOutSlot(epoch);
+    uint256 slotToCompare = _getFlowInSlot(epoch);
+
+    _addFlow(flowLimit_, slotToAdd, slotToCompare, flowOutAmount_);
+}
+```
+    
+### `_addFlowIn(uint256)`
+
+Adds to the flow in amount, also ensuring adherence to the flow limit.
+    
+```solidity
+/**
+ * @notice Adds a flow in amount.
+ * @param flowInAmount_ The flow in amount to add.
+ */
+function _addFlowIn(uint256 flowInAmount_) internal {
+    uint256 flowLimit_ = flowLimit();
+    if (flowLimit_ == 0) return;
+
+    uint256 epoch = block.timestamp / EPOCH_TIME;
+    uint256 slotToAdd = _getFlowInSlot(epoch);
+    uint256 slotToCompare = _getFlowOutSlot(epoch);
+
+    _addFlow(flowLimit_, slotToAdd, slotToCompare, flowInAmount_);
+}
+```
+
+## Setting and Overriding Flow Limits in ITS
+
+There are several options for setting and overriding flow limits in the Interchain Token Service.
+
+### Fully managed flow limits
+
+If you want Axelar to manage the flow limits for the token and not deal with it yourself:
+- Deploy the token with the `minter` address set to `address(0)`
+
+### Partially managed flow limits
+
+If you want to manage the flow limits together with Axelar:
+- Deploy the token with the `minter` address set to your Deployer account or a multi-sig wallet
+If you want to manage the flow limits alone:
+- Deploy the token with the `minter` address set to your Deployer account
+- Call `tokenManager.removeFlowLimiter(itsAddress)` to remove flow limit management by Axelar
+
+### Unmanaged flow limits
+
+If you don't want anyone to manage mint limits:
```

Comment:
@Olanetsoft What is the difference between a flow limit and a mint limit?

Reply:
This is an oversight; it's still the flow limit I am referring to here, not the mint limit. I have updated it. @ffe9f8